src/databricks/labs/ucx/recon/migration_recon.py

```diff
@@ -0,0 +1,141 @@
+    DataComparator,
+    SchemaComparator,
+    TableIdentifier,
+    DataComparisonResult,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ReconResult:
+    src_schema: str
+    src_table: str
+    dst_catalog: str
+    dst_schema: str
+    dst_table: str
+    schema_matches: bool
+    data_matches: bool
+    schema_comparison: str
+    data_comparison: str
+
+
+class MigrationRecon(CrawlerBase[ReconResult]):
+    def __init__(
+        self,
+        sbe: SqlBackend,
+        schema: str,
+        migration_status_refresher: MigrationStatusRefresher,
+        table_mapping: TableMapping,
+        schema_comparator: SchemaComparator,
+        data_comparator: DataComparator,
+        default_threshold: float,
+    ):
+        super().__init__(sbe, "hive_metastore", schema, "recon_result", ReconResult)
+        self._migration_status_refresher = migration_status_refresher
+        self._table_mapping = table_mapping
+        self._schema_comparator = schema_comparator
+        self._data_comparator = data_comparator
+        self._default_threshold = default_threshold
+
+    def snapshot(self) -> Iterable[ReconResult]:
+        return self._snapshot(self._try_fetch, self._crawl)
+
+    def _crawl(self) -> Iterable[ReconResult]:
+        self._migration_status_refresher.reset()
+        migration_index = self._migration_status_refresher.index()
+        migration_rules = self._table_mapping.load()
+        tasks = []
+        for source in migration_index.snapshot():
+            migration_status = migration_index.get(*source)
+            if migration_status is None:
+                continue
+            if not migration_status.dst_catalog or not migration_status.dst_schema or not migration_status.dst_table:
+                continue
+            source_table = TableIdentifier(
+                "hive_metastore",
+                migration_status.src_schema,
+                migration_status.src_table,
+            )
+            target_table = TableIdentifier(
+                migration_status.dst_catalog,
+                migration_status.dst_schema,
+                migration_status.dst_table,
+            )
+            row_comparison = False
+            reconciliation_threshold = self._default_threshold
+            for rule in migration_rules:
+                if rule.match(source_table):
+                    row_comparison = rule.row_comparison
+                    reconciliation_threshold = rule.reconciliation_threshold
+                    break
+            tasks.append(
+                partial(
+                    self._recon,
+                    source_table,
+                    target_table,
+                    row_comparison,
+                    reconciliation_threshold,
+                )
+            )
+        recon_result, errors = Threads.gather("Reconciling data", tasks)
+        if len(errors) > 0:
+            logger.error(f"Detected {len(errors)} while reconciling data")
```

Comment:
can you also show migration errors widget? e.g. we have log errors/warnings widget for assessment, but not for the migration yet.